print-lisp.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2015 Ritesh Ranjan (r.ranjan789@gmail.com)
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote products
+ *    derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * tcpdump filter for LISP - Locator/Identifier Separation Protocol
+ * RFC 6830
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <tcpdump-stdinc.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "ip.h"
+#ifdef INET6
+#include "ip6.h"
+#endif
+
+#include "extract.h"
+#include "interface.h"
+#include "addrtoname.h"
+
+#define IPv4_AFI 1
+#define IPv6_AFI 2
+
+static const char *type_name[] = {
+	"Reserved",
+	"Map-Request",
+	"Map-Reply",
+	"Map-Register",
+	"Map-Notify",
+	"Unallocated-type",
+	"Unallocated-type",
+	"Unallocated-type",
+	"Control-Message",
+};
+
+typedef struct map_register_hdr {
+	uint8_t type_and_flag;
+	uint8_t reserved;
+	uint8_t reserved_and_flag2;
+	uint8_t record_count;
+	uint64_t nonce;
+	uint16_t key_id;
+	uint16_t auth_data_len;
+	uint8_t auth_data[0];
```

Comment:
ISO C forbids zero-size array ('auth_data').
tcpdump is build on many compilers != gcc